---
title: "Figure 2"
author: "Ilaria Cherchi"
date: "2025-02-05"
output: 
  html_document:
    keep_md: yes
---


```{r setup, include=FALSE}
library(data.table)
library(parallel)
library(dplyr)
library(ggplot2)
library(tidyr)
library(ggpubr)
library(MsCoreUtils)
library(ggrepel)
library(purrr)
library(patchwork)
library(GenomicRanges)
library(svglite)
library(org.Hs.eg.db) 
library(KEGGREST)
library(clusterProfiler)
library(Gviz)
library(BSgenome.Hsapiens.UCSC.hg38)
library(BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0)


knitr::opts_chunk$set(dev = c("png", "pdf", "svglite"),
                      fig.ext = c("png", "pdf", "svg"),
                      fig.path = c("Figure_2_S5/"),
                      fig.process = function(filename) {new_filename <- stringr::str_remove(string = filename,
                                        pattern = "-1")
                      fs::file_move(path = filename, new_path = new_filename)
                      ifelse(fs::file_exists(new_filename), new_filename, filename)},
                      cache = F,
                      dpi = 600,
                      unit = "cm",
                      message = FALSE, warning = FALSE)

supp_table_path <- "../Supplementary/"
tables_path <- "../tables/"
rdata_path <- "../rdata/"
utilities_path <- "../utilities/"

source("../utilities/theme_ggplot.R")

palette_seq_mod <- c("Equal" = "gray70", 
                     "Allele swapping" = "#86bfd5",
                     #"Reference mismatch" = "#30622f",
                     #"Indels" = "#7e2453",
                     "[1, 5] mismatches" = "#d25271", 
                      "> 5 mismatches | Indels" = "#7e2453",
                     "Difference in homopolymer" = "#003665" )

palette_homopolymer <- c("Difference in homopolymer" = "#003665", 
                                 "Difference outside homopolymer" = "#29bec3", "Difference proximity" = "gray60")
palette_pairwise <- c("Match" = "gray80", 
                      "Mismatch" = "#cba1ff", 
                      "Deletion" = "#51e6de", 
                      "Insertion" = "#51d454")

chr_t2t <- fread(paste0(utilities_path, "chr_band_t2t.txt"))
chr_hg38 <- fread(paste0(utilities_path, "chr_band_hg38.txt"))
options(ucscChromosomeNames=FALSE)
seqlevelsStyle(BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0) = "UCSC" # Convert chromosome notation to UCSC

```

```{r utils, include = F}
compact_string_to_df <- function(compact_str) {
  # Split by "|" and separate symbols and ranges
  elements <- unlist(strsplit(compact_str, "\\|"))
  
  # Extract symbols and ranges into a dataframe
  df <- do.call(rbind, lapply(elements, function(x) {
    symbol <- substr(x, 1, 1)
    start <- as.numeric(strsplit(substr(x, 2, nchar(x)),"\\(|\\)")[[1]][1])
    end <- as.numeric(strsplit(substr(x, 2, nchar(x)),"\\(|\\)")[[1]][1]) + 
      as.numeric(strsplit(substr(x, 2, nchar(x)),"\\(|\\)")[[1]][2])
    return(data.frame(Symbol = symbol, Start = start, End = end, n_bases = end-start))
  }))
  
  return(df)
}

annotate_df <- function(g1, g2, info_list){
  int <- subsetByOverlaps(g1,g2)
  hits <- findOverlaps(g1,g2)
  for(info in info_list){
    assign(paste(info), unlist(sapply(split(g2@elementMetadata@listData[[info]][subjectHits(hits)],
                                   queryHits(hits)), function(x) paste(unlist(x),collapse="|"))))
    mcols(int) <- DataFrame(mcols(int), mget(info))
  }
  return(int)
}
Sys.setlocale("LC_CTYPE", "en_US.UTF-8")

annotate_gene_list <- function(genes) {
  gene_entrez <- bitr(genes,
                    fromType = "SYMBOL",
                    toType = "ENTREZID",
                    OrgDb = org.Hs.eg.db)


  gene2pathway <- keggLink("pathway", "hsa")
  annot <- gene2pathway[names(gene2pathway) %in% paste0("hsa:", gene_entrez$ENTREZID)]

  gene_to_kegg <- data.frame(
  ENTREZID = sub("hsa:", "", names(annot)),
  PathwayID = sub("path:", "", annot)
  )

  pathway_names <- keggList("pathway", "hsa")  # returns named vector

  pathway_df <- data.frame(
    PathwayID = sub("path:", "", names(pathway_names)),
    PathwayName = as.character(pathway_names),
    stringsAsFactors = FALSE
  )

  gene_to_kegg$PathwayID <- sub("path:", "", gene_to_kegg$PathwayID)

  gene_to_kegg_named <- merge(gene_to_kegg, pathway_df, by = "PathwayID", all.x = TRUE)

  final_annot <- merge(gene_entrez, gene_to_kegg_named, by = "ENTREZID", all.x = TRUE)
  return(final_annot)
}

plot_sequence_top <- function(bs_genome, chrom, pos, width, chr_bands = chr_hg38){
  sTrack <- SequenceTrack(bs_genome)

  gTrack <- GenomeAxisTrack()
  
  colnames(chr_bands) <- c("chrom", "chromStart", "chromEnd", "name", "gieStain")
  
  ideoTrack <- IdeogramTrack(genome = "hg38c", bands = chr_bands, chromosome = chrom, col = "#a12d69")

  displayPars(gTrack) <- list( ticksAt = c(pos-width, pos, pos + width),
                             littleTicks = F,
                             labelPos = "below",
                             lwd = 0.3,
                             cex = 1.2
                             )
  
  displayPars(sTrack) <- list(  lwd = 0.05,
                              min.width = 0,
                              showTitle = T,
                              cex = 1,
                              fontsize = 18,
                              lwd.border.title = 0,
                              col.title = "black")
  displayPars(ideoTrack) <- list(cex = 1.3, lwd = 2)

 names(sTrack) <- "hg38"


  htrack <- HighlightTrack(trackList = list(sTrack), start = pos, end = pos, chromosome = chrom,
                         inBackground = F, fill = NA, col = "#a12d69", lwd = 0.7)

  plotTracks(list(ideoTrack, gTrack, htrack), from = pos - width , to = pos + width, 
           chromosome = chrom,
           margin = c(0, 0, 0, 0) ,
           sizes=c(1, 1.5, 2), add = T)
}

plot_sequence_bottom <- function(bs_genome, chrom, pos, width, chr_bands = chr_t2t){
  colnames(chr_bands) <- c("chrom", "chromStart", "chromEnd", "name", "gieStain")
  ideoTrack <- IdeogramTrack(genome = "hs1", bands = chr_bands, chromosome = chrom, col = "#a12d69")

  sTrack <- SequenceTrack(bs_genome)

  gTrack <- GenomeAxisTrack()
  
  displayPars(ideoTrack) <- list(cex = 1.3, lwd = 2)
  
  displayPars(gTrack) <- list( ticksAt = c(pos-width, pos, pos + width),
                             littleTicks = F,
                             labelPos = "above",
                             lwd = 0.3,
                             cex = 1.2)
  
  displayPars(sTrack) <- list(  lwd = 0.05,
                              min.width = 0,
                              showTitle = T,
                              cex = 1,
                              fontsize = 18,
                              lwd.border.title = 0,
                              col.title = "black")

 names(sTrack) <- "T2T-CHM13 "

 htrack <- HighlightTrack(trackList = list(sTrack), start = pos, end = pos, chromosome = chrom,
                         inBackground = F, fill = NA, col = "#a12d69", lwd = 0.7)

  plotTracks(list( htrack, gTrack, ideoTrack), from = pos - width , to = pos + width, 
           chromosome = chrom,
           margin = c(0, 0, 0, 0) ,
           sizes=c(2,1.5,1), add = T)
}
```

# Figure 2

## Figure 2A
Left side: sketch

For right side load data:

```{r}
all_tracks <- fread(paste0(tables_path, "summary_tracks.tsv"))
all_tracks <- all_tracks %>% 
  mutate(Symbol = factor(Symbol, levels = rev(c("Match", "Mismatch", "Deletion", "Insertion")))) %>% 
  filter(Symbol != "Match") 
```


```{r}
fig_f_df <- all_tracks %>% filter(comparison == "hg38vsT2T") %>%
          mutate(total_bases = (EVENT/prop_bases_event),
                 perc_event_num = as.numeric(sub("%", "", perc_peaks_event)))
fig_f_df$comparison <- "hg38vsT2T-CHM13"
```

```{r figure2A, fig.width=5.8, fig.height=4.8}
(ggplot(fig_f_df, 
        aes( y = Symbol, x = perc_event_num, fill = Symbol))
  + geom_col()
  + scale_fill_manual(values = palette_pairwise,
                      breaks = c("Match", "Mismatch", "Deletion", "Insertion"))
  + theme_publication()
  + facet_grid(track ~ comparison,  switch = "y", space = "free_x")
  + labs (y = "", x = "Sequences with\nat least one modification (%)", fill = "Alignment outcome")
  + geom_label(data = . %>% filter(Symbol == "Mismatch" ), 
               aes(label = perc_peaks_event ), hjust = 1.1, size = 10/.pt, show.legend = F)
  + geom_label(data = . %>% filter(Symbol != "Mismatch" ), 
               aes(label = perc_peaks_event ), hjust = -0.1, size = 10/.pt, show.legend = F)
  + theme(legend.position = "left", legend.justification = "bottom", 
          axis.text.y = element_blank(), axis.ticks.y = element_blank())
)
```


## Figure 2B
Load data:

```{r, warning = F}
load(paste0(rdata_path, "clinvar.RData"))
def <- de
def$CLNREVSTAT <- sub(".*CLNREVSTAT=([^;]+).*", "\\1", def$INFO)

def$cv <- ifelse(def$cv == "Other" & def$class == "Likely_benign", "Likely benign", as.character(def$cv))
def$cv <- ifelse(is.na(def$cv), "Other", as.character(def$cv))

levels(def$cv) <- c("Benign", "Likely benign", "Other", "Conflicting interpretations", "Uncertain significance", "Likely pathogenic", "Pathogenic")

def <- def %>% mutate(hp_status = case_when((hm_match == "F" | is.na(hm_match)) & aln_hg38vsT2T != "." & !is.na(stretch) ~ "Difference outside homopolymer", 
                                              hm_match == "T"  & !is.na(stretch) ~ "Difference in homopolymer",
                                              is.na(stretch) & aln_hg38vsT2T != "." ~ "Difference outside homopolymer",
                                              is.na(stretch)  & aln_hg38vsT2T == "." ~ "Identical proximity",
                                              !is.na(stretch) & aln_hg38vsT2T == "." ~ "Identical proximity", 
                                              .default = NA)) %>%
  mutate(stretch = ifelse(is.na(stretch), "No homopolymer stretch", stretch),
         hp_status = factor(hp_status, levels = c( "Difference in homopolymer",  "Difference outside homopolymer", "Identical proximity")))


def_barplot <- def %>% distinct(ID, nc, cv, stretch, hm_match, 
                                aln_hg38vsT2T, pos.hg38, pos.T2T, chr, gene_info, hp_status,
                                disease, st_dbsnp, type,
                                var_type, CLNREVSTAT) %>% mutate(ID.y = as.numeric(ID)) 

def_barplot$dep <- sub(".*\\((\\d+)\\).*", "\\1", def_barplot$aln_hg38vsT2T)
```

```{r figure2B, fig.width = 11, fig.height = 4.5}
def_barplot$cv <- factor(def_barplot$cv, levels =  c("Benign", "Likely benign", "Other", "Conflicting interpretations", "Uncertain significance", "Likely pathogenic", "Pathogenic"))

dat <- def_barplot %>% 
  mutate(nc = ifelse(hp_status == "Difference in homopolymer", 
                                          "Difference in homopolymer", as.character(nc))) %>%
  group_by(cv) %>% mutate(tot = n()) %>% ungroup() %>% 
  summarize(perc = n() / unique(tot), n= n(), tot = unique(tot), .by = c("cv", "nc"))

dat$nc <- factor(dat$nc, levels = rev(names(palette_seq_mod)))

(ggplot(dat %>% filter(nc != "Equal"), aes( y = cv, x = perc))
              + geom_col(aes(fill = nc))
              + labs(y = "ClinVar classification", 
                     x = "Proportion of ClinVar IDs with differences\nin their proximal sequence", 
                     fill = "Proximal sequence status (±150bp) for hg38vsT2T-CHM13")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom")
              + scale_fill_manual(values = palette_seq_mod, breaks = names(palette_seq_mod))
              + theme(legend.title = element_text(size = 14),
                      legend.margin = margin(0,10,0,-120))
              + guides(fill = guide_legend(nrow = 1, title.position = "top"))
              + scale_x_continuous(labels = scales::percent, limits = c(0,0.4))
              + geom_text(data = . %>% group_by(cv) %>% summarize(perc = sum(perc), tot = unique(tot), n = sum(n)),
                          aes(x = perc, 
                              label = paste0(scales::percent(round(perc, 2)),
                              " (", n, "/", tot, ")")
                                             ),
                          vjust = 0, hjust = -0.4)
              + geom_label(data = . %>% filter(round(perc, 2)*100 > 0), 
                           aes(label = paste0(#"N=",n, "\n", 
                             scales::percent(round(perc, 2))) , x = perc, fill = nc), 
                                 position=position_stack(vjust = 0.5), 
                                 color = "white", size = 3, 
                           show.legend = F,  segment.color = "white")
)
```

## Figure 2C

Extract percentage of difference in pathogenic:
```{r perc}
def %>% mutate(ab = ifelse(nc == "Equal", "Equal", "Sequence modification")) %>%
  group_by(cv,ab) %>%  
  summarise(n = n(), .groups = 'drop') %>% 
  group_by(cv) %>%  
  mutate(percentage =  scales::percent((n / sum(n)))) %>% 
  ungroup() %>% filter(cv=="Pathogenic")

patho_diff <- def %>% filter(cv %in% c("Pathogenic")) 
```

Load COSMIC actionable genes:

```{r actionable}
actionable_genes <- fread(paste0(utilities_path, "Actionability_AllData_v14_GRCh37.tsv")) %>% 
  distinct(GENE) %>% pull(GENE)

length(actionable_genes[actionable_genes %in% patho_diff$gene_info])
```

Compute pathogenic percentages per gene:

```{r perce}
patho_focus <- patho_diff %>% filter(#var_type == "SNV", 
  gene_info %in% actionable_genes)
patho_actio <- patho_focus %>% group_by(gene_info) %>% 
                        summarize(n_IDs = n(), 
                                  n_mod = length(nc[nc != "Equal" & nc != "Allele swapping" & nc != "Reference mismatch"]),
                        prop_diff = n_mod / n_IDs)

patho_actio %>% filter(n_IDs > 10) %>% arrange(desc(prop_diff))
```

Build and annotate barplot:
```{r}
df_barplot_patho <- patho_actio %>% 
          filter(n_IDs > 10, prop_diff > 0.05, n_mod > 20) %>%
          mutate(div = case_when(prop_diff > 0.2 ~ "a",
                                 #prop_mism > 0.30 ~ "b",
                                 .default = "c"))
gene_annot <- annotate_gene_list(df_barplot_patho %>% pull(gene_info))
df_barplot_patho <- left_join(df_barplot_patho, gene_annot %>% distinct(SYMBOL, PathwayName) %>%
                                dplyr::rename(gene_info = SYMBOL, pathway = PathwayName))

df_barplot_patho <- df_barplot_patho %>% ungroup() %>% summarize(n=n(), 
path = case_when(any(grepl("cancer", pathway)) ~ "Pathway in cancer", 
                 .default = head(gsub(" -.*", "", pathway), 1)),
                         .by = c("gene_info", "prop_diff" , "n_mod", "div"))
```

```{r fig2C, fig.height = 8, fig.width = 7.5}
(ggplot(df_barplot_patho, 
 aes(x = reorder(gene_info, - prop_diff), 
     y = round(prop_diff, 2)*100))
 + geom_col(aes(fill = path))
 + theme_publication()
 # + scale_fill_gradient(high = "#132B43",
 #  low = "#56B1F7",
 # )
 # + scale_color_manual(values = c("Actionable gene" = "#a8216b", "Other" = "gray50"))
              
 + geom_label(aes(label = paste0("N=", n_mod) , y = 0), size = 3,
              angle = 90, hjust = -0.1, fill = "transparent", show.legend = F, color = "white")     
 + facet_wrap(div ~ ., scales = "free", ncol = 1 )
 + labs(x = "Actionable genes included in ClinVar", 
        y = "Pathogenic SNVs with differences\nin their proximal sequence (%)",
        fill = "KEGG pathway hits")
  + scale_fill_manual(values = c("#f7b6d3", "#d6a9c8", "#9467bd", "#c5b0d5", "#17becf", "#aec7e8", "#1f77b4"), 
                    na.value = "gray80")
 + scale_color_manual(values = c("#f7b6d3", "#d6a9c8", "#9467bd", "#c5b0d5", "#17becf", "#aec7e8", "#1f77b4"),
                      na.value = "gray80")
  + geom_text(aes(label = scales::percent(prop_diff, 2)), size = 4, vjust = -0.5)
  + scale_y_continuous(expand = expansion(mult = c(0, .1)))
 + theme(
         strip.background = element_blank(),
         strip.text = element_blank(),
         legend.position = "bottom",
         legend.title.position = "top",
         legend.margin = margin(0,0,0,-20),
         axis.text.x = element_text(size = 11),
         #axis.text.x = element_text(angle = 35, hjust = 1, vjust = 1),

         #axis.text.x = element_blank(),
         axis.ticks.x = element_blank()
         #axis.title.x = element_blank()
         )
  + guides(fill = guide_legend(nrow = 3))
)
```

## Figure 2D

```{r disease}
def_ds <- def_barplot %>% filter(!is.na(disease), 
                         !is.na(nc), 
                         disease != "not specified", cv %in% c("Pathogenic", "Likely pathogenic")) 
tp <- def_ds%>% 
   distinct(hp_status, disease, ID, cv) %>% group_by(disease) %>% mutate(tot = n()) %>% ungroup() %>% 
   group_by(hp_status, disease) %>% summarize(N = n(), perc = n()/unique(tot), tot = unique(tot))
order <- tp %>% group_by(disease) %>% summarize(
  perc_var = sum(perc[hp_status != "Identical proximity"]),
  perc_equal = perc[hp_status == "Identical proximity"]) %>%
  arrange(perc_var) %>% pull(disease)

tp <- tp %>% mutate(disease = factor(disease, levels = order))

tp <- tp %>% filter(!(hp_status == "Identical proximity" & perc == 1) & tot > 5 & hp_status != "Identical proximity")

```

```{r figure2D, fig.width = 14, fig.height = 7}

(ggplot(tp %>% filter(N>1), 
        aes( x = perc, y = disease))
  + geom_col( aes(fill = hp_status))
  + theme_publication()
  + scale_fill_manual(values = palette_homopolymer, 
                      breaks = c("Identical proximity","Difference outside homopolymer", "Difference in homopolymer" ))
  + scale_x_continuous(limits = c(-0.1, 0.63),expand = c(0, 0))
  + labs(x = "Proportion of pathogenic and likely pathogenic variants\nwith differences in their proximal sequence", 
         y = "", fill = "Proximal sequence status (±150bp) for hg38vsT2T-CHM13")
   + geom_label(aes(label = paste0("N=",N) , x = perc, fill = hp_status), 
                                 position=position_stack(vjust = 0.5), 
                                 color = "white", size = 3, 
                                  show.legend = F,  segment.color = "white")
 + geom_text(data = . %>% group_by(disease) %>% summarize(perc = sum(perc)),
                          aes(x = perc, 
                              label = scales::percent(round(perc, 2), 1)),
                          vjust = 0.5, hjust = -0.5, size = 4)
 + scale_x_continuous(expand =  expansion(mult = c(0, .1)))
   + theme(
          legend.position = "bottom",
          legend.margin = margin(l = -.45, unit = "npc"))
  )
```

# Supplementary 5

## Supplementary 5A

```{r figureS5A, fig.width = 12, fig.height = 4}
dat <- def_barplot %>% group_by(cv) %>% mutate(hp_class = 
          case_when(hp_status == "Difference in homopolymer" ~ "Difference in homopolymer",
                    stretch == "Homopolymer stretch" ~ "Homopolymer",
                    .default = "No homopolymer")) %>% mutate(tot = n()) %>% ungroup() %>% 
  summarize(perc = n() / unique(tot), n= n(), tot = unique(tot), .by = c("cv", "hp_class"))

(ggplot(dat, aes( y = cv, x = perc))
              + geom_col(aes(fill = hp_class))
              + labs(y = "ClinVar classification", 
                     x = "Proportion of ClinVar IDs with differences\nin their proximal sequence for hg38vsT2T-CHM13", 
                     fill = "")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom")
              + scale_fill_manual(values = c("Difference in homopolymer" = "#003665", 
                                             "Homopolymer" = "#8bad0d", "No homopolymer" = "gray60"), 
                      breaks = c("No homopolymer","Homopolymer", "Difference in homopolymer" ))
 
              + theme(#axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1),
                       legend.title = element_text(size = 14),
                      legend.margin = margin(0,10,0,-120))
              + guides(fill = guide_legend(nrow = 1, title.position = "left"))
              + scale_x_continuous(expand =  expansion(mult = c(0, .15)))
              #         legend.text=element_text(size=22), legend.title = element_text(size=24))
              # + geom_text(data = . %>% group_by(cv) %>% summarize(perc = sum(perc)),
              #             aes(x = perc, 
              #                 label = scales::percent(round(perc, 2), 1)),
              #             vjust = 0, hjust = -0.5)
  + geom_label(aes(label = paste0("N=",n, "\n", scales::percent(round(perc, 2), 1)) , x = perc, fill = hp_class), 
                                 position=position_stack(vjust = 0.5), 
                                 color = "white", size = 3, 
                           #direction = "x",
                            #      max.overlap = 0,
                                  show.legend = F,  segment.color = "white")
              
)
```


## Supplementary 5B

Load ClinVar annotation for hg19vsT2T and hg19vsgh38
```{r, warnings = F, echo = F}
suppressWarnings(load(paste0(rdata_path, "clinvar_classification_hg19vsT2T.RData")))
suppressWarnings(load(paste0(rdata_path, "clinvar_classification_hg19vshg38.RData")))
```

```{r figureS5B, fig.width = 18, fig.height = 7}
def_hg19vsT2T$cv <- ifelse(def_hg19vsT2T$cv == "Other" & def_hg19vsT2T$class == "Likely_benign", "Likely benign", as.character(def_hg19vsT2T$cv))
def_hg19vsT2T$cv <- ifelse(is.na(def_hg19vsT2T$cv), "Other", as.character(def_hg19vsT2T$cv))
def_hg19vsT2T$cv <- factor(def_hg19vsT2T$cv, levels =  c("Benign", "Likely benign", "Other", "Conflicting interpretations", "Uncertain significance", "Likely pathogenic", "Pathogenic"))

def_hg19vshg38$cv <- ifelse(def_hg19vshg38$cv == "Other" & def_hg19vshg38$class == "Likely_benign", "Likely benign", as.character(def_hg19vshg38$cv))
def_hg19vshg38$cv <- ifelse(is.na(def_hg19vshg38$cv), "Other", as.character(def_hg19vshg38$cv))
def_hg19vshg38$cv <- factor(def_hg19vshg38$cv, levels =  c("Benign", "Likely benign", "Other", "Conflicting interpretations", "Uncertain significance", "Likely pathogenic", "Pathogenic"))

a <- (ggplot(def %>% filter(!is.na(nc)), 
                     aes( x = cv, fill = nc))
              + geom_bar()
              + labs(x = "ClinVar classification", y = "N ClinVar IDs", fill = "Sequence alignment")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom",
                         axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1),
                      plot.title = element_text(size = 15)
                      )
               + scale_fill_manual(values = palette_seq_mod)
)
b <- (ggplot(def_hg19vsT2T %>% filter(!is.na(nc)), 
                     aes( x = cv, fill = nc))
              + geom_bar()
              + labs(x = "ClinVar classification", y = "N ClinVar IDs", fill = "Sequence alignment")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom",
                          axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1),
                       plot.title = element_text(size = 15)
                      )
              + scale_fill_manual(values = palette_seq_mod)
)
c <- (ggplot(def_hg19vshg38, 
                     aes( x = cv, fill = nc))
              + geom_bar(show.legend = F)
              + labs(x = "ClinVar classification", y = "N ClinVar IDs", fill = "Sequence alignment")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom",
                        axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1),
                       plot.title = element_text(size = 15)
                      )
              + scale_fill_manual(values = palette_seq_mod)
             )

patchwork::wrap_plots(a + labs(title = "hg38vsT2T-CHM13", x = "") , b + labs(title = "hg19vsT2T-CHM13", y = ""), 
                      c + labs(title = "hg19vshg38", y = "", x = ""),
                      nrow= 1, widths = c(4,4,4), guides = "collect", axes = "collect") & plot_annotation(theme = theme(legend.position = 'bottom'))

# (a + labs(title = "hg38vsT2T-CHM13", x = "")) + (b + labs(title = "hg19vsT2T-CHM13", y = "")) + (c + labs(title = "hg19vshg38", y = "", x = "")) + plot_layout(nrow = 1, widths = c(4,4,4), guides = "collect", axes = "collect") & theme(legend.position = 'bottom')

```

## Supplementary 5C

```{r select}
selected <- def %>% filter(gene_info %in% df_barplot_patho$gene_info)
```

```{r figureS5C, fig.width = 18, fig.height = 7}
(ggplot(selected, 
        aes( x = reorder(gene_info, pos.hg38), fill = nc))
              + geom_bar(position = "fill")
              + labs(x = "Gene", y = "Proportion of ClinVar IDs", fill = "Sequence alignment hg38vsT2T-CHM13")
              + theme_publication()
              + theme(panel.grid = element_blank(), legend.position = "bottom",
                       axis.text.x = element_text(angle = 25, hjust = 1, vjust = 1)
             )
              + facet_grid(.~ chr, scales = "free", space = "free")
              + scale_fill_manual(values = palette_seq_mod)
)
```

## Supplementary 5D

```{r}
id <-  405391
pos_t2t <- def$pos.T2T[def$ID == id]
pos_hg38 <- def$pos.hg38[def$ID == id]
pos_hg19 <- def$pos.hg19[def$ID == id]
chrom <- def$chr[def$ID == id]
gene <- def$gene_info[def$ID == id]

width <- 150

def[def$ID == id]
```

```{r figureS5D, fig.width = 18, fig.height = 4}
grid.newpage()
pushViewport(viewport(layout = grid.layout(3, 1, heights=c(0.2, 0.5,0.5))))
pushViewport(viewport(layout.pos.row = 1, layout.pos.col = 1))
grid.text(paste0("ClinVar ID: ", id, " - Gene: ", gene), gp = gpar(fontsize = 16, fontface = "bold"))
grid.legend(labels = c("A","C","T","G"), nrow = 1,
           pch = 15,  # square symbols
           gp = gpar(col = c("#9cd0e3", "#ea9249", "#2466a8", "#cd0016"), cex = 0.8),
           hgap = unit(0.5, "lines"),
           vp = viewport(x = 0.85, y = 0.5))
popViewport(1)  
pushViewport(viewport(layout.pos.col = 1, layout.pos.row = 2))
plot_sequence_top(BSgenome.Hsapiens.UCSC.hg38, chrom, pos_hg38, width)
grid.lines(x = unit(c(0.014, 0.014), "npc"),
           y = unit(c(0.03, 0.97), "npc"),
           gp = gpar(col = "black", lwd = 2))
popViewport(1)
pushViewport(viewport(layout.pos.col = 1, layout.pos.row = 3))
plot_sequence_bottom(BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0, chrom, pos_t2t, width)
grid.lines(x = unit(c(0.014, 0.014), "npc"),
           y = unit(c(0.03, 0.97), "npc"),
           gp = gpar(col = "black", lwd = 2))

popViewport(1)
```

